File: android/src/main/java/org/ligi/passandroid/ui/edit/ColorPickDialog.java
package org.ligi.passandroid.ui.edit;

import android.content.Context;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import com.larswerkman.holocolorpicker.ColorPicker;
import org.greenrobot.eventbus.EventBus;
import org.ligi.passandroid.R;
import org.ligi.passandroid.events.PassRefreshEvent;
import org.ligi.passandroid.model.pass.Pass;

public class ColorPickDialog {

    public static void showColorDialog(final Context context, final Pass pass, final EventBus bus) {
        final View inflate = LayoutInflater.from(context).inflate(R.layout.edit_color, null);

        final ColorPicker colorPicker = (ColorPicker) inflate.findViewById(R.id.colorPicker);
        colorPicker.setColor(pass.getAccentColor());
        colorPicker.setOldCenterColor(pass.getAccentColor());
        new AlertDialog.Builder(context).setView(inflate).setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(final DialogInterface dialog, final int which) {
                pass.setAccentColor(colorPicker.getColor());
                bus.post(new PassRefreshEvent(pass));
            }
        }).setNegativeButton(android.R.string.cancel, null).setTitle(R.string.change_color_dialog_title).show();
    }
}

File: android/src/main/java/org/ligi/passandroid/ui/edit/dialogs/ColorPickDialog.kt
package org.ligi.passandroid.ui.edit.dialogs

import android.content.Context
import android.view.LayoutInflater
import androidx.appcompat.app.AlertDialog
import org.ligi.passandroid.R
import org.ligi.passandroid.databinding.EditColorBinding
import org.ligi.passandroid.model.pass.Pass

fun showColorPickDialog(context: Context, pass: Pass, refreshCallback : () -> Unit) {
    val inflate = EditColorBinding.inflate(LayoutInflater.from(context))

    inflate.colorPicker.color = pass.accentColor
    inflate.colorPicker.oldCenterColor = pass.accentColor
    AlertDialog.Builder(context).setView(inflate.root).setPositiveButton(android.R.string.ok) { _, _ ->
        pass.accentColor = inflate.colorPicker.color
        refreshCallback.invoke()
    }.setNegativeButton(android.R.string.cancel, null).setTitle(R.string.change_color_dialog_title).show()
}
